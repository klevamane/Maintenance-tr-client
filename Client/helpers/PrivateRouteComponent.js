import React from 'react';
import { connect } from 'react-redux';
import { Route, Redirect} from 'react-router-dom';
import PropTypes from 'prop-types';


const PrivateRouteComponent = ({ component: Component, auth, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      // check to see if authenticated, if true then load the component else redirect to login
      (auth.isAuthenticated === true ? (<Component {...props} />) : (<Redirect to="/" />))}
    />
);

PrivateRouteComponent.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
});
export default connect(mapStateToProps)(PrivateRouteComponent);
