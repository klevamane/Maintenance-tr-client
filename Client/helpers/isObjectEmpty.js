
const isObjectEmpty = (value) => {
  if (value === undefined || (typeof value === 'object' && Object.keys(value).length === 0)
  ) return true;
};
export default isObjectEmpty;
