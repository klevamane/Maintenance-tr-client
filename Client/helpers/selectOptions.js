export const selectOptionsFault = [

  { label: 'Select one', value: '' },
  { label: 'Broken screen', value: 'Broken screen' },
  { label: 'Water damage', value: 'Water damage' },
  { label: 'Unlocking Operator', value: 'Unlocking Operator' },
  { label: 'Screen Issues', value: 'Screen Issues' },
  { label: 'Proximity Sensor', value: 'Proximity Sensor' },
  { label: 'Front camera', value: 'Front camera' },
  { label: 'Rear camera', value: 'Rear camera' },
  { label: 'Keyboard', value: 'Keyboard' },
  { label: 'Flash', value: 'Flash' },
  { label: 'Battery replacement', value: 'Battery replacement' },
  { label: 'network issues', value: 'network issues' },
  { label: 'Lagging', value: 'Lagging' },
];

export const selectOptionRepair = [
  { label: 'Select one', value: '' },
  { label: 'Maintenance', value: 'maintenance' },
  { label: 'Repair', value: 'repair' },
];
