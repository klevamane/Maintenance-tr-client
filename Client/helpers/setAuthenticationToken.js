import axios from 'axios';

export const instance = axios.create({
  baseURL: process.env.BACKEND_URL,
  timeout: 10000,
});

const setAuthenticationToken = (token) => {
  if (token) {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
  } else {
    delete axios.defaults.headers.common['Authorization'];
  }
};

export default setAuthenticationToken;
