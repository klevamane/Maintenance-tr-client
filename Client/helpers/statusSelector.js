const StatusSelector = (status) => {
  let statusIndicator;
  switch (status) {
    case 'Disapproved':
      statusIndicator = 'status-pill-rejected';
      break;
    case 'Pending':
      statusIndicator = 'status-pill-pending';
      break;
    case 'Resolved':
      statusIndicator = 'status-pill-accepted';
      break;
    default:
      statusIndicator = 'status-pill-inactive';
  }
  return statusIndicator;
};

export default StatusSelector;
