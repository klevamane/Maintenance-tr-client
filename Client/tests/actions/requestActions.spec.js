import moxios from 'moxios';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { instance as api } from '../../helpers/setAuthenticationToken';
import {
  getAnyUserRequestById, getEveryUsersRequests, approveArequest,
  disapproveArequest, resolveApproveOrDisapprove, getArequest,
  getUserRequests, createRequest,
} from '../../actions/requestActions';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {};
const store = mockStore(initialState);
beforeEach(() => {
  moxios.install(api);
});

afterEach(() => {
  moxios.uninstall(api);
});


const requestDetails = {
  fault: 'screen issues',
  description: 'request description',
  brand: 'Benq',
};

describe('Actions: Request Actions', () => {
  beforeEach(() => { // Runs before each test in the suite
    store.clearActions();
  });

  it('should test the action to set request loader', () => {
    // Dispatch action
    store.dispatch(getAnyUserRequestById(6));

    const actions = store.getActions();
    const requestLoading = [{ type: 'REQUEST_LOADING' }];
    const expectedPayload = requestLoading;
    expect(actions).toEqual(expectedPayload);
  });

  it('should successfully dispatch the getAnyUserRequestById action', async (done) => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: [],
      });
    });
    const expectedTypes = [{ type: 'GET_ARTICLE_SEARCH' }];
    store.dispatch(getAnyUserRequestById(1));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });
  it('should successfully dispatch the getRequest  action', async (done) => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: [],
      });
    });

    const expectedTypes = [{ type: 'VIEW_EVERY_USERS_REQUESTS' }];
    store.dispatch(getEveryUsersRequests());
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });
  
  it('should successfully dispatch the approveArequest action', async (done) => {
    const expectedTypes = [{ type: 'APPROVE_REQUEST' }];
    store.dispatch(approveArequest(80));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });

  it('should unsuccessfully dispatch the approveArequest action', async (done) => {
    const expectedTypes = [{ type: 'GET_ERRORS' }];
    store.dispatch(approveArequest('hv'));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });

  it('should successfully dispatch the disapproveArequest action', async (done) => {
    const expectedTypes = [{ type: 'DISAPPROVE_REQUEST' }];
    store.dispatch(disapproveArequest(80));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });

  it('should successfully dispatch the resolveApproveOrDisapprove action', async (done) => {
    const expectedTypes = [{ type: 'CHANGE_REQUEST_STATUS' }];
    store.dispatch(resolveApproveOrDisapprove(80, 'disapprove'));
    expect(store.getActions()).toMatchSnapshot();
    // setTimeout(() => {
    // //   expect(store.getActions()[0]).toEqual(expectedTypes);
    //   expect(store.getActions()[0]).toMatchSnapshot();
    //   done();
    // }, 2000);
    done();
  });

  it('should successfully dispatch the getArequest action', async (done) => {
    const expectedTypes = [{ type: 'VIEW_REQUEST' }];
    store.dispatch(getArequest(80));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });

  it('should successfully dispatch the getUserRequests action', async (done) => {
    const expectedTypes = [{ type: 'VIEW_USER_REQUESTS' }];
    store.dispatch(getUserRequests())
      .then(() => {
        expect(store.getActions()[0]).toEqual(expectedTypes);
      });
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });

  it('should successfully dispatch the createRequest action', async (done) => {
    const expectedTypes = [{ type: 'CREATE_REQUEST' }];
    const history = '/';
    store.dispatch(createRequest(requestDetails, history));
    setTimeout(() => {
      expect(store.getActions()[0]).toEqual(expectedTypes);
      done();
    }, 2000);
    done();
  });
});
