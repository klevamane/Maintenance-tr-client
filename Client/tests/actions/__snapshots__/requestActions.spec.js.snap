// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions: Request Actions should successfully dispatch the resolveApproveOrDisapprove action 1`] = `
Array [
  Object {
    "type": "REQUEST_LOADING",
  },
]
`;
